src/doc-utils.js

```diff
@@ -120,6 +125,7 @@ function breakParentGroup(groupStack) {
 }
 
 function propagateBreaks(doc) {
+  var alreadyVisited = new WeakMap();
```

Comment:
Theoretically this could just be a `Map`, right? This goes out of scope anyway and will be GCed so the strong reference will go away anyway.